### ISSUE TITLE
Minitest with Knapsack

### ISSUE THREAD

**Issue by paulca** [buildkite/test-collector-ruby#111]
Hi,

I just installed the test collector in our app, following the instructions on the site.

I ran in to an initial issue that `WebMock` wasn't allowing requests, and then I needed to move the `WebMock` `allow` config above `Buildkite::TestCollector.configure(hook: :minitest)`.

This got the tests running, but now I'm getting an error in the internals:

```

# Running:
--
  |  
  | Capybara starting Puma...
  | * Version 5.6.4 , codename: Birdie's Version
  | * Min threads: 0, max threads: 4
  | * Listening on http://172.16.112.5:3000
  | rake aborted!
  | TypeError: no implicit conversion of Pathname into String
  | /bundle/ruby/2.7.0/gems/buildkite-test_collector-1.0.0/lib/buildkite/test_collector/minitest_plugin/trace.rb:60:in `delete_prefix'
  | /bundle/ruby/2.7.0/gems/buildkite-test_collector-1.0.0/lib/buildkite/test_collector/minitest_plugin/trace.rb:60:in `file_name'
  | /bundle/ruby/2.7.0/gems/buildkite-test_collector-1.0.0/lib/buildkite/test_collector/minitest_plugin/trace.rb:53:in `location'
  | /bundle/ruby/2.7.0/gems/buildkite-test_collector-1.0.0/lib/buildkite/test_collector/minitest_plugin/trace.rb:40:in `as_hash'
  | /bundle/ruby/2.7.0/gems/buildkite-test_collector-1.0.0/lib/buildkite/test_collector/session.rb:145:in `write_result'
  | /bundle/ruby/2.7.0/gems/buildkite-test_collector-1.0.0/lib/buildkite/test_collector/minitest_plugin/reporter.rb:15:in `record'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:874:in `block in record'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:873:in `each'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:873:in `record'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:350:in `run_one_method'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:337:in `block (2 levels) in run'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:336:in `each'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:336:in `block in run'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:376:in `on_signal'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:363:in `with_info_handler'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:335:in `run'
  | /bundle/ruby/2.7.0/gems/railties-6.1.6/lib/rails/test_unit/line_filtering.rb:10:in `run'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:169:in `block in __run'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:169:in `map'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:169:in `__run'
  | /bundle/ruby/2.7.0/gems/minitest-5.15.0/lib/minitest.rb:146:in `run'
  | /bundle/ruby/2.7.0/gems/knapsack_pro-3.2.1/lib/knapsack_pro/runners/queue/minitest_runner.rb:98:in `minitest_run'
  | /bundle/ruby/2.7.0/gems/knapsack_pro-3.2.1/lib/knapsack_pro/runners/queue/minitest_runner.rb:72:in `run_tests'
  | /bundle/ruby/2.7.0/gems/knapsack_pro-3.2.1/lib/knapsack_pro/runners/queue/minitest_runner.rb:32:in `run'
  | /bundle/ruby/2.7.0/gems/knapsack_pro-3.2.1/lib/tasks/queue/minitest.rake:6:in `block (3 levels) in <main>'
  | /bundle/ruby/2.7.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
  | /usr/local/bin/bundle:113:in `load'
  | /usr/local/bin/bundle:113:in `<main>'
  | Tasks: TOP => knapsack_pro:queue:minitest
  | (See full trace by running task with --trace)

```

Is it because we're using Knapsack? Seems reasonable that a path would be a `Pathname` rather than a string.

It's Rails 6.1.4 on Ruby 2.7.4

Thanks!

**Comment by nickvanw** [comment]
I ran into this as well, the fix seems to be just calling `#to_s` on `Rails.root` here: https://github.com/buildkite/test-collector-ruby/blob/main/lib/buildkite/test_collector/minitest_plugin/trace.rb#L69

`Dir.getwd` returns a String, `Rails.root` returns a Pathname.

**Comment by paulca** [comment]
@nickvanw Did you monkey-patch for now? Shall I create a PR?

**Comment by paulca** [comment]
[PR](https://github.com/buildkite/test-collector-ruby/pull/112)

**Comment by JuanitoFatas** [comment]
Thanks @paulca @nickvanw for the reports and fix 💜

I’ve released [v1.0.1](https://rubygems.org/gems/buildkite-test_collector/versions/1.0.1) that includes the fix.
